File: 07_Polymorphism/Run-TIme_Polymorphism/Main.java
class Animal {
    void sound() {
        System.out.println("This is an animal sound.");
    }
}

class Dog extends Animal {
    @Override
    void sound() {
        System.out.println("The dog barks.");
    }
}

class Cat extends Animal {
    @Override
    void sound() {
        System.out.println("The cat meows.");
    }
}

public class Main {
    public static void makeSound(Animal animal) {
        animal.sound(); 
    }

    public static void main(String[] args) {
        Animal dog = new Dog();
        Animal cat = new Cat();

        makeSound(dog); 
        makeSound(cat); 
    }
}

File: 07_Polymorphism/Run-TIme_Polymorphism/Main.cpp
#include <iostream>
using namespace std;

class Animal {
public:
    virtual void sound() {
        cout << "This is an animal sound." << endl;
    }
};

class Dog : public Animal {
public:
    void sound(string str) {
        cout << "This sound i need to make " << str << endl;
    }
};

class Cat : public Animal {
public:
    void sound() override {
        cout << "The cat meows." << endl;
    }
};

void makeSound(Animal* animal) {
    animal->sound(); 
}

int main() {
    Dog dog;
    Cat cat;

    makeSound(&dog); 
    makeSound(&cat); 

    return 0;
}
